evm/db/state.py

```diff
@@ -44,13 +53,24 @@ class State:
 
     logger = logging.getLogger('evm.state.State')
 
-    def __init__(self, db, root_hash=BLANK_ROOT_HASH, read_only=False):
+    def __init__(
+        self,
+        db,
+        root_hash=BLANK_ROOT_HASH,
+        read_only=False,
+        read_list=None,
+        write_list=None
+    ):
         if read_only:
             self.db = ImmutableDB(db)
         else:
             self.db = db
         self._trie = HashTrie(Trie(self.db, root_hash))
 
+        self.check_access_restrictions = read_list is not None or write_list is not None
```

Comment:
*naming nitpick*

What do you think about changing this property to `is_access_restricted`